golang/cosmos/app/upgrade.go

```diff
@@ -92,6 +92,10 @@ func unreleasedUpgradeHandler(app *GaiaApp, targetUpgrade string) func(sdk.Conte
 			// Each CoreProposalStep runs sequentially, and can be constructed from
 			// one or more modules executing in parallel within the step.
 			CoreProposalSteps = []vm.CoreProposalStep{
+				vm.CoreProposalStepForModules(
+					// Upgrade Zoe (no new ZCF needed).
+					"@agoric/builders/scripts/vats/upgrade-zoe.js",
```

Comment:
I was going to suggest an incarnation number test, but it looks like we already have one:

https://github.com/Agoric/agoric-sdk/blob/735a449fd15495e2c31c481693c3f0b24f8b11a6/a3p-integration/proposals/n%3Aupgrade-next/initial.test.js#L12

It seems likely that this will fail in ci and need updating. But in any case, it's there.